src/main/java/com/aws/greengrass/util/EncryptionUtils.java

```diff
@@ -129,17 +129,41 @@ private static KeyPair readPkcs8PrivateKey(byte[] pkcs8Bytes) throws GeneralSecu
     }
 
     private static KeyPair readPkcs1PrivateKey(byte[] pkcs1Bytes) throws GeneralSecurityException {
-        // We can't use Java internal APIs to parse ASN.1 structures, so we build a PKCS#8 key Java can understand
+        // We can't use Java internal APIs to parse ASN.1 structures, so we build a PKCS#8 key Java
+        // can understand
         int pkcs1Length = pkcs1Bytes.length;
         int totalLength = pkcs1Length + 22;
-        // reference to https://github.com/Mastercard/client-encryption-java/blob/master/src/main/java/com/mastercard/developer/utils/EncryptionUtils.java#L95-L100
+        // reference to
+        // https://github.com/Mastercard/client-encryption-java/blob/master/src/main/java/com/mastercard/developer/utils/EncryptionUtils.java#L95-L100
         // this method can save us from importing BouncyCastle as dependency
-        byte[] pkcs8Header = {0x30, (byte) 0x82, (byte) ((totalLength >> 8) & 0xff), (byte) (totalLength & 0xff),
+        byte[] pkcs8Header = {0x30,
+                (byte) 0x82,
+                (byte) ((totalLength >> 8) & 0xff),
+                (byte) (totalLength & 0xff),
                 // Sequence + total length
-                0x2, 0x1, 0x0, // Integer (0)
-                0x30, 0xD, 0x6, 0x9, 0x2A, (byte) 0x86, 0x48, (byte) 0x86, (byte) 0xF7, 0xD, 0x1, 0x1, 0x1, 0x5, 0x0,
+                0x2,
+                0x1,
+                0x0, // Integer (0)
+                0x30,
+                0xD,
+                0x6,
+                0x9,
+                0x2A,
+                (byte) 0x86,
+                0x48,
+                (byte) 0x86,
+                (byte) 0xF7,
+                0xD,
+                0x1,
+                0x1,
+                0x1,
+                0x5,
+                0x0,
                 // Sequence: 1.2.840.113549.1.1.1, NULL
-                0x4, (byte) 0x82, (byte) ((pkcs1Length >> 8) & 0xff), (byte) (pkcs1Length & 0xff)
+                0x4,
+                (byte) 0x82,
+                (byte) ((pkcs1Length >> 8) & 0xff),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is using a bit-wise shifting operation. Incorrect bit-wise shifting operations may lead to overflow or underflow errors. Consider checking the value of the operand before the shifting operation and check that the amount by which the operand is shifted does not lead to overflow.